bpf/cpu/cpu.bpf.c

```diff
@@ -573,14 +575,15 @@ static __always_inline bool retrieve_task_registers(u64 *ip, u64 *sp, u64 *bp) {
 // right now using both.
 static __always_inline bool has_fp(u64 current_fp) {
   u64 next_fp;
+  u64 ra;
 
   for (int i = 0; i < MAX_STACK_DEPTH; i++) {
     int err = bpf_probe_read_user(&next_fp, 8, (void *)current_fp);
+    bpf_probe_read_user(&ra, 8, (void *)current_fp + 8);
```

Comment:
not sure, but in all the verifier errors I read something about `invalid read from stack off -120+0 size 8` ... 

1) should we not be checking the returned error here?
2) are we sure the `current_fp + 8` is correct?

Reply:
> should we not be checking the returned error here?

Yes, definitely (this should not be resulting in the invalid reads, though)

> are we sure the current_fp + 8 is correct?

It should, in x86 the return address is always 8 bytes ahead from the current frame pointer